lib/plugins/progress.js

```diff
@@ -1,25 +1,22 @@
 export default (shower, options) => {
-    const containerElement = shower.container.getElement();
     const progressSelector = options.selector || '.progress';
-    const bar = containerElement.querySelector(progressSelector);
+    const bar = shower.container.querySelector(progressSelector);
     if (!bar) return;
 
     bar.setAttribute('role', 'progressbar');
     bar.setAttribute('aria-valuemin', 0);
     bar.setAttribute('aria-valuemax', 100);
 
+    const { player } = shower;
     const updateProgress = () => {
-        const current = shower.player.getCurrentSlideIndex();
-        if (current === -1) return;
-
-        const total = shower.getSlidesCount();
-        const progress = current / (total - 1) * 100;
+        const index = player.currentSlideIndex;
+        const { length } = player.slides;
+        const progress = index / (length - 1) * 100;
 
         bar.style.width = `${progress}%`;
         bar.setAttribute('aria-valuenow', progress);
         bar.setAttribute('aria-valuetext', `Slideshow progress: ${progress}%`);
     };
 
-    shower.player.events.on('activate', updateProgress);
-    updateProgress();
```

Comment:
Now we emit events (like `modechange` on initial shower load) after plugins are inited, so authors would not have to make extra calls and checks.